products/sle/main.pm

```diff
@@ -35,6 +35,138 @@ use known_bugs;
 
 init_main();
 
+package SchedulerFactory{
+    use Data::Dumper ;    
+    
+    sub new {
+	my ($class, $testname) = @_;
+	#print "attrs --> $attrs{'testname'}";
+	my $self = { 
+	    testname_factory => TestSuiteFactory::getTestSuite($testname)
+	};
+	bless $self, $class;
+	return $self;    
+    };
+    
+    sub createScheduler {
+	my ($self) = shift;
+	$self->{testname_factory};
+    }
+}
+
+
+package TestSuiteFactory {
+    my %sleTests = (
+	boot_linuxrc => sub { Boot_linuxrc->new('boot_linuxrc')},
+	installer_extended_textmode => sub {InstallerExtendedTextmode->new('installer_extended_textmode')}
+	);
+    sub getTestSuite{
+	my ($name) = shift;
+	print "name is $name\n";
+	return $sleTests{$name}->()
+    }
+}
+
+package TestSuiteInterface {
+    use File::Find;
+    use File::Basename;
+    use main_common;
+    BEGIN {
+	unshift @INC, dirname(__FILE__) . '/../../lib';
+    }
+    sub new {
+	my ($class, $name) = @_;
+	my $self = {};
+	$self->{name} = $name;
+	bless $self, $class;
+	return $self;
+    }
+    sub getVariables {
+	my ($self) = shift;
+	return $self->{variables}
+    }
+    sub loadtests {
+	my ($self) = shift;
+	{
+	    #die "oops" unless $self->{scheduler}
+	    $_->() for (values %{$self->{scheduler}});
+	    return 1;
+	}
+	return 0;
+    }
+}
+
+package InstallerExtendedTextmode {
+    use base 'TestSuiteInterface';
+    use main_common;
+    sub new {
+	my ($class, $name) = @_;
+	my $self = {};
+	
+	$class->SUPER::new($name);
+	$self->{variables}={
+	    CHECK_PRESELECTED_MODULES=>'1',
+	    CHECK_RELEASENOTES=>'1',
+	    EULA_LANGUAGES=>'korean',
+	    EXIT_AFTER_START_INSTALL=>'1',
+	    EXTRABOOTPARAMS=>'Y2STRICTTEXTDOMAIN=1',
+	    INSTALLER_EXTENDED_TEST=>"1",
+	    VALIDATE_CHECKSUM=>"1"
+	};
+	$self->{scheduler} = {
+	    isosize => sub{ loadtest("installation/isosize.pm"); },
```

Comment:
is there chance that value will contain something else than `sub{ loadtest(<path>); }` ? if no than we could simplify it to : 
```suggestion
	    isosize => "installation/isosize.pm",
```

and have loop somewhere after all which do loadtest call for each element

Reply:
i cant say. in short yes, the hash could be easily adapt your suggestion. in my approach I thought is easier this way and then do [this](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/9545/files#diff-aeec25095520b82be463f8f83f49fd62R23)